### ISSUE TITLE
Issues with LibreSSL 2.9.2

### ISSUE THREAD

**Issue by protonesso** [libressl-portable/portable#528]
Hi, I'm using LibreSSL 2.9.2 instead of OpenSSL in my distro. I'm cross-compiling it for x86_64-linux-musl. I use this configuration to compile LibreSSL:
```
./configure --build=x86_64-cross-linux-gnu --host=x86_64-linux-musl --prefix=/usr --with-openssldir=/etc/ssl
```
But while running GNU make I see this error:
```
  CC       compat/reallocarray.lo
compat/getprogname_linux.c: In function 'getprogname':
compat/getprogname_linux.c:32:2: error: #error "Cannot emulate getprogname"
   32 | #error "Cannot emulate getprogname"
      |  ^~~~~
compat/getprogname_linux.c:34:1: warning: control reaches end of non-void function [-Wreturn-type]
   34 | }
      | ^
```

**Comment by steils** [comment]
Maybe you want to change the issue title to a more proper one?:)
Something like "Compilation with non-glibc C libraries broken in LibreSSL 2.9.2".

**Comment by busterb** [comment]
Fixed in #529
